File: PurpleUI-iOS/Sources/PurpleUI/Tokens/PUColors.swift
import SwiftUI

/// Purple UI semantic colour tokens.
///
/// Always reference these in components — never use raw hex values.
/// Dark-mode mappings are noted in comments; swap values when `colorScheme == .dark`.
public struct PUColors {

    // MARK: - Brand
    public static let brand       = Color(hex: "#7458FD")  // invariant light/dark
    public static let brandSubtle = Color(hex: "#eceaff")  // dark: #2a1f6b

    // MARK: - Backgrounds
    public static let background         = Color(hex: "#FFFFFF")  // dark: #011638
    public static let backgroundElevated = Color(hex: "#F7F7F8")  // dark: #0a2048
    public static let backgroundSunken   = Color(hex: "#F9F9FC")  // dark: #09193d
    public static let backgroundNavy     = Color(hex: "#011638")  // invariant
    public static let backgroundAlt      = Color(hex: "#F5F1ED")  // invariant (cream)

    // MARK: - Text / Icons
    public static let onBackground          = Color(hex: "#000000")  // dark: #FFFFFF
    public static let onBackgroundSecondary = Color(hex: "#595959")  // dark: rgba(255,255,255,0.70)
    public static let onBackgroundTertiary  = Color(hex: "#AAACB0")  // dark: rgba(255,255,255,0.35)

    // MARK: - Borders
    public static let outline       = Color(hex: "#CDCED0")  // dark: rgba(255,255,255,0.15)
    public static let outlineSubtle = Color(hex: "#EFF0F0")  // dark: rgba(255,255,255,0.06)

    // MARK: - Misc
    public static let loaderTrack = Color(hex: "#DDDDDF")  // invariant

    // MARK: - Status
    public static let success       = Color(hex: "#16C172")
    public static let successSubtle = Color(hex: "#E8F5E9")  // dark: #0d3320
    public static let successStrong = Color(hex: "#4CAF50")
    public static let error         = Color(hex: "#F03A47")
    public static let errorSubtle   = Color(hex: "#F03A47").opacity(0.12)
    public static let warning       = Color(hex: "#E9D502")
    public static let warningSubtle = Color(hex: "#fefbe6")  // dark: #3d3400
    public static let info          = Color(hex: "#045DEC")
    public static let infoSubtle    = Color(hex: "#045DEC").opacity(0.12)

    // MARK: - Vendor (activity feed only)
    public static let vendorBrown  = Color(hex: "#6B2737")
    public static let vendorForest = Color(hex: "#2D5A27")
    public static let vendorRed    = Color(hex: "#E4002B")
    public static let vendorAmber  = Color(hex: "#F5A623")
}

// MARK: - Hex colour initialiser
public extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3:  (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6:  (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8:  (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default: (a, r, g, b) = (255, 0, 0, 0)
        }
        self.init(
            .sRGB,
            red:     Double(r) / 255,
            green:   Double(g) / 255,
            blue:    Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}

File: PurpleUI-Android/purpleui/src/main/kotlin/ai/purple/purpleui/tokens/PUColors.kt
package ai.purple.purpleui.tokens

import androidx.compose.ui.graphics.Color

/**
 * Semantic color tokens for Purple UI.
 *
 * Values are transcribed 1:1 from the `color` group in `tokens.json` (the
 * design system's single source of truth). Do not hardcode hex values in
 * components - reference these instead.
 *
 * Dark-mode mappings are noted in comments. Implement your own
 * `darkPUColors` object and swap via CompositionLocal for dark mode.
 */
object PUColors {
    // Brand
    val brand       = Color(0xFF7458FD)  // dark: #7458FD (unchanged)
    val brandSubtle = Color(0xFFECEAFF)  // dark: #2a1f6b

    // Backgrounds
    val background          = Color(0xFFFFFFFF)  // dark: #011638 (backgroundNavy)
    val backgroundElevated  = Color(0xFFF7F7F8)  // dark: #0a2048
    val backgroundSunken    = Color(0xFFF9F9FC)  // dark: #09193d
    val backgroundNavy      = Color(0xFF011638)  // invariant - always navy
    val backgroundElevatedNavy = Color(0xFF0A2048)  // invariant - dark-mode-only elevated surface (cards/sheets/inputs raised above backgroundNavy). No light equivalent.
    val backgroundAlt       = Color(0xFFF5F1ED)  // invariant - always cream

    // On-background text / icons
    val onBackground          = Color(0xFF000000)  // dark: #FFFFFF
    val onBackgroundSecondary = Color(0xFF595959)  // dark: rgba(255,255,255,0.70)
    val onBackgroundTertiary  = Color(0xFFAAACB0)  // dark: rgba(255,255,255,0.35)

    // Borders & dividers
    val outline       = Color(0xFFCDCED0)  // dark: rgba(255,255,255,0.15)
    val outlineSubtle = Color(0xFFEFF0F0)  // dark: rgba(255,255,255,0.06)

    // Misc component-specific
    val loaderTrack = Color(0xFFDDDDDF)  // invariant

    // Status
    val success       = Color(0xFF16C172)  // invariant
    val successSubtle = Color(0xFFE8F5E9)  // dark: #0d3320
    val successStrong = Color(0xFF4CAF50)  // invariant

    val error       = Color(0xFFF03A47)
    val errorSubtle = error.copy(alpha = 0.12f)  // rgba(240,58,71,0.12)

    val warning       = Color(0xFFE9D502)
    val warningSubtle = Color(0xFFFEFBE6)  // dark: #3d3400

    val info       = Color(0xFF045DEC)
    val infoSubtle = info.copy(alpha = 0.12f)  // rgba(4,93,236,0.12)

    // Vendor / partner brand colours - Activity feed only, do not use elsewhere
    val vendorBrown  = Color(0xFF6B2737)
    val vendorForest = Color(0xFF2D5A27)
    val vendorRed    = Color(0xFFE4002B)
    val vendorAmber  = Color(0xFFF5A623)

    // --------------------------------------------------------------
    // Android-local tokens: not in tokens.json.
    //
    // These names/values exist only in this file - they carry over from
    // before Android's tokens were properly wired up to the design system
    // and are still referenced by components below. Flagged here as drift
    // to be resolved (either mapped onto a real tokens.json token or
    // removed) rather than silently deleted.
    // --------------------------------------------------------------
    val border          = Color(0xFFE2E8F0)  // Android-local: not in tokens.json. Used by PUSearchBar; likely should be `outline`.
    val borderFocus     = Color(0xFF7458FD)  // Android-local: not in tokens.json. Used by PUSearchBar; value matches `brand`.
    val inputBackground = Color(0xFFF8F8FA)  // Android-local: not in tokens.json. Used by PUSearchBar (light mode).
    val overlay         = Color(0x80011638)  // Android-local: not in tokens.json. Used by PUAlertModal (50% backgroundNavy scrim).
}
